[12.0][FIX] l10n_es_aeat_mod390: diseno de registro 2021

Joan10 opened the pull request:
Mod390 was not working for the new register design for 2021 period (dr209e2021.xlsx). Two fields were renamed in page 5 and 5 more added in page 6. Added new xml files for the new register design and renamed 2019's.

pedrobaeza commented on l10n_es_aeat_mod390/__manifest__.py:
```diff
@@ -49,6 +49,21 @@
+        'data/aeat_export_mod390_2021_sub08_data.xml',
+        'data/aeat_export_mod390_2021_main_data.xml',
+        'data/tax_code_map_mod390_data.xml',
+        'views/mod390_view.xml',
+        'security/ir.model.access.csv',
+        'security/l10n_es_aeat_mod390_security.xml',
```
Esto no hay que repetirlo

pedrobaeza commented on l10n_es_aeat_mod390/data/aeat_export_mod390_2019_main_data.xml:
```diff
@@ -4,7 +4,7 @@
 
     <record id="aeat_mod390_2019_main_export_config" model="aeat.model.export.config">
-        <field name="name">Mod.390 2019-actualidad</field>
+        <field name="name">Mod.390 2019</field>
```
Pon la fecha donde finalizó. Yo diría que el formato no es 2021, si no 2022, aunque sea para presentar el 390 del 2021, por lo que debería ser:
```suggestion
        <field name="name">Mod.390 2019-2021</field>
```

pedrobaeza commented on l10n_es_aeat_mod390/data/aeat_export_mod390_2021_main_data.xml:
```diff
@@ -0,0 +1,225 @@
+<odoo>
+
+    <record id="aeat_mod390_2021_main_export_config" model="aeat.model.export.config">
+        <field name="name">Mod.390 2021-actualidad</field>
```
```suggestion
        <field name="name">Mod.390 2022-actualidad</field>
```
y cambiar el nombre de los archivos

pedrobaeza commented on l10n_es_aeat_mod390/data/aeat_export_mod390_2019_main_data.xml:
```diff
@@ -4,7 +4,7 @@
 
     <record id="aeat_mod390_2019_main_export_config" model="aeat.model.export.config">
-        <field name="name">Mod.390 2019-actualidad</field>
+        <field name="name">Mod.390 2019-2021</field>
```
Otra cosa: aquí faltaría la fecha fin `2021-12-31`

HaraldPanten commented:
Hola @Joan10 

Primero de todo, quiero agradecerte el hecho estar atento a los cambios y por haber hecho esta aportación.

¿Tienes disponibilidad para aplicar el cambio que te comenta Pedro?

Un saludo.

xAdrianC-Kernet commented:
Hola, estoy revisando un poco, sobre todo desde un punto de vista técnico por lo que a nivel contable mi opinión no cunenta demasiado, pero me da la sensación de que el registro aeat_mod390_2019_main_export_config deberia tener date_end 2020-12-31 en lugar de 2021-12-31 como se propone y el registro aeat_mod390_2021_main_export_config el name Mod.390 2021-actualidad.
Ya que es el nuevo registro que se está introduciendo es el que tiene vigencia para el ejercicio 2021 y si le ponemos la fecha final 2021-12-31 al que vamos a dejar obsoleto, cuando calculamos el ejercicio 2021 sigue tirando del anterior.

¿Cómo lo veis @HaraldPanten @pedrobaeza?

pedrobaeza commented:
Si el nuevo diseño de registro afecta al 390 que hay que presentar ahora del 2021, entonces efectivamente debe ser la fecha fin del otro `2020-12-31`.

HaraldPanten commented:
Hola @xAdrianC-Kernet 

Todas las aportaciones son valorables. Esto es una Comunidad y cuanto más participativos sean los usuarios, más acaba repercutiendo positivamente en el software.

Por lo que comentas, en mi opinión tiene sentido el cambio.

anajuaristi commented:
Hola!
Lo hemos testeado funcionalmente LGTM :+1: 
Ya direis si hace falta algo más para que se pueda mergear.

pedrobaeza commented:
Los comentarios están ya hechos y esperando respuesta.

Joan10 commented:
Disculpad, primera PR que hago y no sé si estáis pendientes de que yo haga algo más. Tengo que añadir yo el cambio propuesto por @xAdrianC-Kernet? O no es necesario para aprobar la PR?

xAdrianC-Kernet commented:
hola @Joan10 yo entiendo que faltarían por incluir los cambios en el data del manifest, que te propone @pedrobaeza, ya que estás repitiendo ficheros y lo que yo te comento en el campo date_end en el fichero aeat_mod390_2019_main_export_config.

En cuanto incluyas esas dos modificaciones en el commit de tu rama actual (GotesSCoop:12.0)  se actualizan automaticamente en este PR

HaraldPanten commented:
> Disculpad, primera PR que hago y no sé si estáis pendientes de que yo haga algo más. Tengo que añadir yo el cambio propuesto por @xAdrianC-Kernet? O no es necesario para aprobar la PR?

Hola Joan, normalmente el flujo es el siguiente:

1- Un usuario realiza un PR. (En este caso tú)

2- Los demás usuarios hacen las revisiones técnicas y funcionales (en este caso Pedro, Adrián, Ana o yo mismo). Durante este proceso, pueden surgir dudas o cambios sugeridos que el autor del PR debería seguir (siempre que sean sugerencias correctas) o debatir/proponer una solución alternativa si no le encaja lo que los demás usuarios le comentan. Por ejemplo: los cambios sugeridos por Pedro, o el error que detectó Adrián en las fechas.

Estos cambios deberías realizarlos tú, ya que eres quien abrió el PR. También puede realizarlos algún compañero tuyo que tenga permisos en vuestro rama/repositorio para hacerlo.

Los demás usuarios estábamos a la espera de tus respuestas/cambios.

3- Una vez los cambios/dudas están resueltos, el PR se va validando por los usuarios que lo estaban revisando. En este caso tanto Adrián como Ana lo han validado.

4- Una vez tiene suficientes revisiones, el PR está listo para fusionarse por parte de un PSC del repositorio, o por otro usuario con permisos suficientes en el repositorio (OCA Board Members, normalmente).

En este caso concreto, Pedro debería validar si sus comentarios se han tenido en cuenta.

Por lo que yo veo, deberías cambiar el mensaje del último commit y especificar un poco más los cambios que has realizado. No te olvides de poner un [FIX] delante del mensaje.

Creo que no me he dejado nada... ¡Gracias por colaborar!

HaraldPanten commented:
Disculpa Joan, no lo había detectado antes. Creo que hay algo raro en tus commits.

Por un lado debería haber el [FIX] del cambio de diseño, y por otra parte el [FIX] del cambio de fechas. Así diferenciamos cada uno de los cambios.

Lo que no me cuadra son los otros 2 commits. Creo que deberías hacer squash de forma que únicamente te queden los 2 correspondientes. Te adjunto captura:

<img width="1345" alt="Captura de pantalla 2022-01-26 a las 12 21 30" src="https://user-images.githubusercontent.com/32449357/151155039-98638c2f-eb96-4bfa-ba7c-3f4d498d759a.png">

pedrobaeza commented:
Todo debería ser un solo commit, ya que es un único cambio lógico.

HaraldPanten commented:
> Todo debería ser un solo commit, ya que es un único cambio lógico.

Pues uno, entonces, y simplificamos.

Joan10 commented:
Solamente le di a aceptar a los commits de Pedro. No hice nada más. 

He intentado hacer squash pero creo que aún lo he complicado más. Ya diréis qué más tengo que hacer.

Mis pasos:
git rebase -i HEAD~4  # Juntados los 4 commits. Primero en pick y los 3 siguientes en squash.
git pull origin 12.0 # ya que no me dejaba hacer push
git push origin 12.0 

Ahora no sé como ha quedado eso...

Entiendo que el commit bueno es el [aba6b68fd](https://github.com/OCA/l10n-spain/pull/2012/commits/aba6b68fd99d6428281860795f681691a047ca72)

xAdrianC-Kernet commented:
yo creo que el problema ha sido hacer el hacer el pull, yo en ese caso, despues del rebase y juntar y quitar los commits que corresponda, haria un push --force, ya que no hay peligro de conflictos al solo trabajar tú en esa rama

Commit:
[FIX] l10n_es_aeat_mod390: diseno de registro 2021

Mod390 was not working for the register design dr209e2021. Changes commited:
 - Two fields were renamed in page 5 and 5 more added in page 6.
 - Start and end dates for 2019 model were also changed from 2019 to 2020.
 - Consequently, new model date starts at 2021 until now.

Joan10 commented:
Vale, creo que ahora sí. Podéis confirmar? Gracias.

Al final he ido 2 commits atrás (hasta el último del fork) y con los cambios no trackeados he hecho un nuevo commit. He forzado un push después a mi rama.

```
git reset --soft HEAD~2
git commit
git push --force origin 12.0
```

xAdrianC-Kernet commented:
yo ahora lo veo perfecto, los cambios implementados y un solo commit, peor soy un superviviente también! que confirmen si el formato se adecua ahora @HaraldPanten y @pedrobaeza

HaraldPanten reviewed (approved):
Yo lo veo OK

Patrickkappert commented:
Hola, gracias para hacer los cambios por el formato cambiado.  Para cuando esta previsto el merge? (si se espera un par de dias mas ya no hace falta... ) ;)  
saludos

pedrobaeza commented:
@Patrickkappert puedes utilizar este PR en cualquier momento sin necesidad de hacer merge, o arreglarlo por tu cuenta.

Joan10 commented:
Se puede hacer merge ya o tengo que hacer algo más? @pedrobaeza

pedroortega-punt commented:
@pedrobaeza Como está este pull para mergear?, el lunes es el último día para presentar. Gracias

HaraldPanten commented:
Hola,

Tened en cuenta que podéis colaborar con vuestras revisiones técnicas también. @vipvalen @Patrickkappert 

Si técnicamente os encaja, yo mismo lo fusionaré para descargar un poco a Pedro. De momento veo los comentarios, pero no veo validaciones 😝.

Tomaré en cuenta la revisión de Ana como usuaria con experiencia contrastada.

A la espera de las validaciones. Saludos.

anajuaristi commented:
@HaraldPanten agradezco mucho el comentario pero.. mi experiencia personal es más constatada en otras áreas. Contabilidad en general no es que me entusiasme demasiado (por decirlo suave) 
Esta área mejor con Olatz. Ella sí tiene experiencia constatada y puede probar cualquier PR o lo que necesiteis tanto en runbot como en cualquier otro entorno de test, a nivel funcional. Cualquier cosa que necesiteis de ella me decís, porque no suele entrar en github.
Ya direis!!
:)

pedroortega-punt commented:
@HaraldPanten @pedrobaeza , probado este pull en un cliente real y todo ok. Se puede mergear

HaraldPanten commented:
> @HaraldPanten @pedrobaeza , probado este pull en un cliente real y todo ok. Se puede mergear

Hola,

Una revisión debería hacerse del siguiente modo, te lo dejo muy muy resumido:

1- Ve a la pestaña de files changed:
<img width="1351" alt="Captura de pantalla 2022-01-27 a las 17 22 24" src="https://user-images.githubusercontent.com/32449357/151401120-e8232699-f16d-4958-a6aa-786085df92df.png">


2- Haz clic en "Review changes":
<img width="1361" alt="Captura de pantalla 2022-01-27 a las 17 23 38" src="https://user-images.githubusercontent.com/32449357/151401152-fc2d00c2-ffdd-4723-84fe-0cbdd455e6bf.png">


3- Escoge si quieres añadir comentarios, pedir cambio o si quieres aprobar el PR.
4- Normalmente se añade un comentario con tus observaciones, etc.
5- Clic en Submit review:

<img width="760" alt="Captura de pantalla 2022-01-27 a las 17 23 47" src="https://user-images.githubusercontent.com/32449357/151401188-2b1b3a1d-013d-478e-97d2-3f83452c146d.png">


Venga, va. Me espero a que lo valides para fusionar, depende de ti, @vipvalen ! 😝

pedroortega-punt reviewed (commented):
Probado y subido correctamente a la aeat en un cliente real.

pedroortega-punt commented:
@HaraldPanten Gracias por la explicación, ya está el comentario realizado.

HaraldPanten commented:
> @HaraldPanten Gracias por la explicación, ya está el comentario realizado.

Casi casi @vipvalen . Si se trata de una validación en la que para ti está todo OK, tienes que hacer "Approve". En tu revisión no has hecho "Approve" sino "Comment". Te paso captura para que veas la que tienes que escoger:
<img width="737" alt="Captura de pantalla 2022-01-27 a las 17 54 40" src="https://user-images.githubusercontent.com/32449357/151406228-99056db9-f7fc-434d-884e-2dc6e5075e1d.png">

pedroortega-punt reviewed (approved):
Probado en cliente real y subido a la aeat correctamente.

pedroortega-punt commented:
@HaraldPanten gracias y perdona, esto es nuevo para mi

pedrobaeza commented:
Venga, vamos con ello:

/ocabot merge major

Ahora después lo pondré en v13 y v14, que quiero pasarlo a archivos CSV como hicimos en el 303 para mayor comodidad de manejo y velocidad de carga.

HaraldPanten commented:
> @HaraldPanten gracias y perdona, esto es nuevo para mi

No hay problema, siempre que haya buena actitud y ánimo de colaborar, no sabe mal echar un cable en lo que se pueda 👍

HaraldPanten commented:
> Venga, vamos con ello:
> 
> /ocabot merge major
> 
> Ahora después lo pondré en v13 y v14, que quiero pasarlo a archivos CSV como hicimos en el 303 para mayor comodidad de manejo y velocidad de carga.

Perfecto! Merci Pedro

pedrobaeza commented:
Pues aquí tenéis otro parche que también es necesario: #2058

xAdrianC-Kernet commented:
Gracias a todos, me encargo yo de llevarlo a la v11 para los que quedamos con algún rezagado

rafaelbn commented:
Hola @OCA/local-spain-maintainers , estamos viendo que no funciona el 390 en 15, y creemos que es por este PR que no está en 13, 14 y 15.

Me podéis confirmar y lo llevamos nosotros a 15?

Muchas gracias!!🤗

pedrobaeza commented:
Sí que está esto en versiones superiores. Ejemplo para v13: https://github.com/OCA/l10n-spain/commit/11d74d5c473cbcfcef15eee3d4a48b05d194ab7f

Lo que ocurre es que hay nuevo formato BOE para el 390 para 2022: https://sede.agenciatributaria.gob.es/Sede/ayuda/disenos-registro/modelos-300-399.html

rafaelbn commented:
¡Gracias! Aclarado. Hay que adaptar los modelos por tanto. 👍
